lib/ClassUtils.lua

```diff
@@ -29,6 +29,32 @@ function ClassUtils.makeClass(name, constructor)
 		setmetatable(SubClass, {__index = self})
 		return SubClass
 	end
+	function Class:extendWithInterface(name, interface)
+		local function getComposableInterface(input)
+			if input == nil then
+				return function()
+					return {}
+				end
+			elseif type(input) == "function" then
+				return input
+			else
+				return function()
+					return input
+				end
+			end
+		end
+		local inheritedInterface = self.interface
+		local compositeInterface = function(Class)
+			return TableUtils.assign(
+				{},
+				getComposableInterface(interface)(Class),
+				getComposableInterface(inheritedInterface)(Class)
```

Comment:
What's the expected behaviour if `interface` and `inherited` interface have overlapping keys? If I extend a class A and make the interface for the subclass B less restrictive, I can now create instances of A that don't adhere to A's interface. Do we care?

Reply:
Good spot. Not sure it matters too much, but I'll add a comment